packages/services/src/kernel/restapi.ts

```diff
@@ -235,3 +235,122 @@ export async function getKernelModel(
   validateModel(data);
   return data;
 }
+
+/**
+ * The kernel API client.
+ *
+ * #### Notes
+ * Use this class to interact with the Jupyter Server Kernel API.
+ * This class adheres to the Jupyter Server API endpoints.
+ */
+export class KernelAPIClient {
+  /**
+   * Create a new kernel API client.
+   *
+   * @param options - The options used to create the client.
+   */
+  constructor(options: { serverSettings?: ServerConnection.ISettings } = {}) {
+    this.serverSettings =
+      options.serverSettings ?? ServerConnection.makeSettings();
+  }
+
+  /**
+   * The server settings for the client.
+   */
+  readonly serverSettings: ServerConnection.ISettings;
+
+  /**
+   * List the running kernels.
+   *
+   * @returns A promise that resolves with the list of running kernel models.
+   *
+   * #### Notes
+   * Uses the [Jupyter Server API](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/jupyter-server/jupyter_server/main/jupyter_server/services/api/api.yaml#!/kernels) and validates the response model.
+   *
+   * The promise is fulfilled on a valid response and rejected otherwise.
+   */
+  async listRunning(): Promise<IModel[]> {
+    return listRunning(this.serverSettings);
+  }
+
+  /**
+   * Get a kernel model.
+   *
+   * @param id - The id of the kernel of interest.
+   *
+   * @returns A promise that resolves with the kernel model.
+   *
+   * #### Notes
+   * Uses the [Jupyter Server API](https://petstore.swagger.io/?url=https://raw.githubusercontent.com/jupyter-server/jupyter_server/main/jupyter_server/services/api/api.yaml#!/kernels) and validates the response model.
+   *
+   * The promise is fulfilled on a valid response and rejected otherwise.
+   * If the kernel does not exist on the server, the promise is resolved with `undefined`.
+   */
+  async getKernelModel(id: string): Promise<IModel | undefined> {
```

Comment:
Not an opinioon yet, just a question, have you thought whether these names should be like `getModel` and `restart` over `getKernelModel` and `restartKernel`?

Reply:
Good point.

There is also `startNew` which does not have `kernel` in its name.

I guess it would make sense to drop the `kernel` part since the methods are already under the kernel client.

Reply:
For now these were mostly mappings to the existing functions already accessible from `KernelAPI`. There could also be a point of having the same function/method names when using `KernelAPI` and `KernelClientAPI`.

Reply:
Happy to push a new commit with renamed methods, so we can decided which way would be preferable.